WebRoot/js/zimbraAdmin/common/ZaTree.js

```diff
@@ -524,10 +524,22 @@ function(item, ev) {
 
         this._selectedItems.add(item);
 
-		if (item._setSelected(true)) {
-            this._updateHistory(item, true, isShowInHistory);
-			this._notifyListeners(DwtEvent.SELECTION, [item], DwtTree.ITEM_SELECTED, ev, this._selEv);
-		}
+        if (item._setSelected(true)) {
+            if (item._text === ZaMsg.NAD_Tab_HSM) {
+                if (typeof ZaHSM === 'undefined') {
+                    ZaApp.getInstance().getCurrentController().popupErrorDialog(ZaMsg.ERROR_Feature_Not_Licensed + "<br>" + ZaMsg.INFO_gotoLicenseet);
+                }
+                if (ZaHSM && ZaHSM.checkstoragefeaturenabled()) {
+                    this._updateHistory(item, true, isShowInHistory);
+                    this._notifyListeners(DwtEvent.SELECTION, [item], DwtTree.ITEM_SELECTED, ev, this._selEv);
+                } else {
+                    ZaApp.getInstance().getCurrentController().popupErrorDialog(ZaMsg.ERROR_Feature_Not_Licensed + "<br>" + ZaMsg.INFO_gotoLicense);
+                }
```

Comment:
when `ZaHSM` is undefined won't it visit both: line 530 and 536? Not sure if that is what you intend to do.

Reply:
you are correct I missed to add the else if block.Added it